provider/k8s.go

```diff
@@ -1215,17 +1217,69 @@ func (mat FileStoreMaterialization) IterateSegment(begin, end int64) (FeatureIte
 	}, nil
 }
 
+func (mat FileStoreMaterialization) NumChunks() (int, error) {
+	searchPath, err := mat.store.CreateFilePath(fileStoreResourcePath(mat.id))
+	if err != nil {
+		return -1, err
```

Comment:
convert -1 to a const. easy to fat finger a diff value between the 8 uses.

Reply:
Yeah but its actually the error that should be checked, the -1 is to hopefully force some sort of error if you forget to check the err

Reply:
If this was a special value that meant error, I'd agree, but I just arbritrarily decided -1 instead of 0

Reply:
it's less about the error checking logic and more on the syntax consistency of using a sentinel value in multiple spots. Like if someone fat fingers `-11` then that could cause a bug if a conditional explicitly checks for -1

ex. `if iter.maxIdx != -1` -> some random explosion or data bug

but if you think it can't happen then you're good to resolve!

Reply:
I think what Simba says makes sense, the  value doesn't matter if there's an error

Reply:
Yeah but shouldn't depend on that value at all, should check the error. I see your point, but I think having a const and exporting it is overkill since it'd have to be documented and would make the behavior fixed.